macros/contexts/contextNondecimalBase.pl

```diff
@@ -0,0 +1,274 @@
+
+    BEGIN_PGML
+    Convert [$a] to base-5:
+
+    [$a] = [__]{$a_5}[`_5`]
+    END_PGML
+
+=cut
+
+sub _contextNondecimalBase_init {
+	context::NondecimalBase::Init(@_);
+	sub convertBase { context::NondecimalBase::convert(@_); }
+}
+
+package context::NondecimalBase;
+
+# Define the contexts 'NondecimalBase' and 'LimitedNondecimalBase'
+sub Init {
+	my $context = $main::context{NondecimalBase} = context::NondecimalBase::Context->new();
+	$context         = $main::context{LimitedNondecimalBase} = $context->copy;
+	$context->{name} = 'LimitedNondecimalBase';
+	$context->operators->undefine($context->operators->names);
+	$context->parens->undefine('|', '{', '[');
+}
+
+=head2 convertBase
+
+The function C<convertBase(value, opts)> converts the value from or to other bases depending on the options
+in C<opts>.  The input C<value> is a positive number or string version of a positive number in some base.
+
+=head3 options
+
+=over
+
+=item * C<from> the base that C<value> is in.  Defaults to 10.
+
+=item * C<to> the base that C<value> is to be converted to.
+
+=item * C<digits> the digits to be used for the conversion.  The default is 0..9, 'A'.. 'E'
+up through hexadecimal.
+
+=back
+
+=head3 Examples
+
+For the following, since C<from> is not used, the base of C<value> is assumed to be 10.
+
+    convertBase(58, to => 5);  # returns 213
+    convertBase(58, to => 8); # returns 72
+    convertBase(734, to => 16); # returns 2DE
+
+For the following, since C<to> is not used, these are converted to base 10.
+
+    convertBase(213, from => 5); # returns 58
+    convertBase(72, from => 8); # returns 58
+    convertBase('2DE', from => 16); # returns 734
+
+Both C<to> and C<from> can be used together.
+
+    convertBase(213, from => 5, to => 8); # returns 72
+
+If one wants to use a different set of digits, say 0..9, 'T', 'E' for base-12 as an example
+
+    convertBase(565, to => [0..9,'T','E']);  # returns '3E1'
+
+=cut
+
+sub convert {
+	my ($value, %options) = @_;
+	my $from = $options{'from'} // 10;
+	my $to   = $options{'to'}   // 10;
+
+	my $context = $main::context{NondecimalBase}->copy;
+	if ($from != 10) {
+		$context->setBase($from);
+		$value = $context->fromBase($value);
+	}
+	if ($to != 10) {
+		$context->setBase($to);
+		$value = $context->toBase($value);
+	}
+	return $value;
+}
+
+package context::NondecimalBase::Context;
+our @ISA = ('Parser::Context');
+
+# Create a Context based on Numeric (Real) that allows +, -, *, / and ^ on nondecimal numbers.
+
+sub new {
+	my $self    = shift;
+	my $class   = ref($self) || $self;
+	my $context = bless Parser::Context->getCopy('Numeric'), $class;
+	$context->{name}           = 'NondecimalBase';
+	$context->{parser}{Number} = 'context::NondecimalBase::Number';
+	$context->{value}{Real}    = 'context::NondecimalBase::Real';
+	$context->functions->disable('All');
+	$context->constants->clear();
+	$self->{pattern}{number}               = '[' . join('', 0 .. 9, 'A' .. 'Z') . ']+';
```

Comment:
1. Even though in general it would be nice to extend to bases larger than 36, I'm wondering if this is needed for WeBWorK problems. 
2. if we add `'a'..'z'` to the `$context->{pattern}{number}`, we should be able to use lower case letter (and then the size of the base could grow to 62). 

I updated the `{pattern}{number}` like you suggested.